src/main/java/wooteco/subway/domain/PathCalculator.java

```diff
@@ -3,58 +3,72 @@
+        Set<Station> stations = extractStations(lines);
+        WeightedMultigraph<Station, ShortestPathEdge> graph = getMultiGraph(stations, lines);
         return new PathCalculator(new DijkstraShortestPath<>(graph));
     }
 
-    private static Set<Station> extractStations(List<Section> sections) {
+    private static Set<Station> extractStations(List<Line> lines) {
         Set<Station> stations = new HashSet<>();
-        for (Section section : sections) {
-            stations.add(section.getUpStation());
-            stations.add(section.getDownStation());
+        for (Line line : lines) {
+            Sections sections = line.getSections();
+            for (Section section : sections.getSections()) {
+                stations.add(section.getUpStation());
+                stations.add(section.getDownStation());
+            }
         }
         return stations;
     }
 
-    private static WeightedMultigraph<Station, DefaultWeightedEdge> getMultiGraph(Set<Station> stations,
-            List<Section> sections) {
-        WeightedMultigraph<Station, DefaultWeightedEdge> graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
+    private static WeightedMultigraph<Station, ShortestPathEdge> getMultiGraph(Set<Station> stations,
+            List<Line> lines) {
+        WeightedMultigraph<Station, ShortestPathEdge> graph = new WeightedMultigraph<>(ShortestPathEdge.class);
         addVertex(stations, graph);
-        addEdge(sections, graph);
+        addEdge(lines, graph);
         return graph;
     }
 
-    private static void addVertex(Set<Station> stations, WeightedMultigraph<Station, DefaultWeightedEdge> graph) {
+    private static void addVertex(Set<Station> stations, WeightedMultigraph<Station, ShortestPathEdge> graph) {
         for (Station station : stations) {
             graph.addVertex(station);
         }
     }
 
-    private static void addEdge(List<Section> sections, WeightedMultigraph<Station, DefaultWeightedEdge> graph) {
-        for (Section section : sections) {
-            graph.setEdgeWeight(graph.addEdge(section.getUpStation(), section.getDownStation()), section.getDistance());
+    private static void addEdge(List<Line> lines, WeightedMultigraph<Station, ShortestPathEdge> graph) {
+        for (Line line : lines) {
+            Sections sections = line.getSections();
+            for (Section section : sections.getSections()) {
+                graph.addEdge(section.getUpStation(), section.getDownStation(), new ShortestPathEdge(line.getId(),
+                        section.getDistance()));
+            }
         }
     }
 
     public List<Station> calculateShortestPath(Station source, Station target) {
         return dijkstraShortestPath.getPath(source, target).getVertexList();
     }
 
+    public List<Long> calculateShortestPathLines(Station source, Station target) {
+        List<ShortestPathEdge> edgeList = dijkstraShortestPath.getPath(source, target).getEdgeList();
```

Comment:
```suggestion
        List<ShortestPathEdge> edges = dijkstraShortestPath.getPath(source, target).getEdgeList();
```

List와 같은 자료형을 이름으로 쓰면 어떤 문제가 있을까요?